flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -1479,6 +1485,72 @@ public void testLateRegisteredPartitionsWillBeReleased() {
         assertThat(shuffleMaster.getExternallyReleasedPartitions(), hasSize(1));
     }
 
+    @Test
+    public void testCheckpointCleanerIsClosedAfterCheckpointServices() throws Exception {
+        final JobGraph jobGraph = singleJobVertexJobGraph(1);
+        enableCheckpointing(jobGraph);
+        final CountDownLatch checkpointServicesShutdownBlocked = new CountDownLatch(1);
+        final CountDownLatch cleanerClosed = new CountDownLatch(1);
```

Comment:
A slightly simpler variant of the `CountDownLatch` is the `OneShotLatch`.

Reply:
I'm aware of that class, but I'm probably missing some intuition behind it. Why is better than using java primitive that does the same thing?